### ISSUE TITLE
The select2('data') method was called on an element that is not using Select2

### ISSUE THREAD

**Issue by gaop** [NejcZdovc/ng2-select2#114]
someone else has the same problem? [version: 1.0.0-beta.11]
![1691517326278_ pic_hd](https://user-images.githubusercontent.com/3616633/35577508-b881e880-061c-11e8-9477-c5137d5f3f61.jpg)

**Comment by gaop** [comment]
I have found out.
The error will be triggered when setting "option.matcher" to do custom filter like this:
```
    options: Select2Options = <Select2Options> {
        ...,
        matcher: (term: string, text: string, option: any) => {
            // do customer filter
        }
    };
```

Finally, I have fix the bug using async & await (ES6):
```
    private async initPlugin() {
        ...
        if(options.matcher) {
            let oldMatcher: any = await this.requireOldMatcher();
            options.matcher = oldMatcher(options.matcher);
            this.element.select2(options);

            if (typeof this.value !== 'undefined') {
                this.setElementValue(this.value);
            }
        }
        ...
    }
    async requireOldMatcher() : Promise<any> {
        return new Promise<any[]>(resolve => {
            jQuery.fn.select2.amd.require(['select2/compat/matcher'], (oldMatcher: any) => {
                resolve(oldMatcher);
            });
        });
    }
```

**Comment by gaop** [comment]
https://github.com/gaop/ng2-select2

**Comment by htrex** [comment]
I'm using the matcher option and seeing the same error message.

**Comment by NejcZdovc** [comment]
fixed with #115 in version 1.0.0-beta.13
